bin/PerlDDS/Run_Test.pm

```diff
@@ -92,25 +92,36 @@ sub formatted_time {
 sub wait_kill {
   my $process = shift;
   my $wait_time = shift;
-  my $desc = shift;
+  my $name = shift;
   my $verbose = shift;
+  my $opts = shift;
+
   $verbose = 0 if !defined($verbose);
 
   my $ret_status = 0;
   my $start_time = formatted_time();
   if ($verbose) {
-    print STDERR "$start_time: waiting $wait_time seconds for $desc before "
+    print STDERR "$start_time: waiting $wait_time seconds for $name before "
       . "calling kill\n";
   }
-  my $result = $process->WaitKill($wait_time);
+
+  my $result = $process->WaitKill($wait_time, $opts);
+
   my $time_str = formatted_time();
   if ($result != 0) {
-      my $ext = ($verbose ? "" : "(started waiting for termination at $start_time)");
-      print STDERR "$time_str: ERROR: $desc returned $result $ext\n";
-      $ret_status = 1;
+    my $ext = ($verbose ? "(started waiting for termination at $start_time)" : "");
+    if ($result == -1) {
+      print STDERR "$time_str: ERROR: $name timedout $ext\n";
+    } elsif ($result == 255) {
+      print STDERR "$time_str: ERROR: $name terminated with a signal $ext\n";
```

Comment:
It'd be great if we could print the signal. It seems to be possible to get the name from the number too: https://stackoverflow.com/questions/29862178/name-of-signal-number-2

Reply:
The signal name should already be printed by PerlACE's`check_return_value` function that is called by this function. Here, we're just kind of summarizing the result with a timestamp.